src/data_gradients/batch_processors/formatters/classification.py

```diff
@@ -0,0 +1,77 @@
+from typing import Tuple, Optional, Callable, List
+
+import torch
+from torch import Tensor
+
+from data_gradients.batch_processors.utils import check_all_integers
+from data_gradients.batch_processors.formatters.base import BatchFormatter
+from data_gradients.batch_processors.formatters.utils import ensure_images_shape, ensure_channel_first, drop_nan
+from data_gradients.config.data.data_config import DetectionDataConfig, ClassificationDataConfig
+from data_gradients.batch_processors.formatters.utils import DatasetFormatError
+
+
+class UnsupportedClassificationBatchFormatError(DatasetFormatError):
+    def __init__(self, str):
+        super().__init__(str)
+
+
+class ClassificationBatchFormatter(BatchFormatter):
+    """Classification formatter class"""
+
+    def __init__(
+        self,
+        data_config: ClassificationDataConfig,
+        class_names: List[str],
+        class_names_to_use: List[str],
+        n_image_channels: int,
+    ):
+        """
+        :param class_names:         List of all class names in the dataset. The index should represent the class_id.
+        :param class_names_to_use:  List of class names that we should use for analysis.
+        :param n_image_channels:    Number of image channels (3 for RGB, 1 for Gray Scale, ...)
+        """
+        self.data_config = data_config
+
+        class_names_to_use = set(class_names_to_use)
+        self.class_ids_to_use = [class_id for class_id, class_name in enumerate(class_names) if class_name in class_names_to_use]
+
+        self.n_image_channels = n_image_channels
+
+    def format(self, images: Tensor, labels: Tensor) -> Tuple[Tensor, Tensor]:
+        """Validate batch images and labels format, and ensure that they are in the relevant format for detection.
+
+        :param images: Batch of images, in (BS, ...) format
+        :param labels: Batch of labels, in (BS) format
+        :return:
+            - images: Batch of images already formatted into (BS, C, H, W)
+            - labels: Batch of targets (BS)
+        """
+
+        images = ensure_channel_first(images, n_image_channels=self.n_image_channels)
+        images = ensure_images_shape(images, n_image_channels=self.n_image_channels)
+        labels = self.ensure_labels_shape(images=images, labels=labels)
+
+        if 0 <= images.min() and images.max() <= 1:
+            images *= 255
+            images = images.to(torch.uint8)
+        elif images.min() < 0:  # images were normalized with some unknown mean and std
```

Comment:
This won't get us the original image though...but maybe its the best we can do ?
I am wondering if in this case, we would rather have a visualizable image that has different pixel values then something else.

Just thinking out loud here, whether users would be confused by the result image from this.

Reply:
Ok, I'm continuing thinking out loud as well. Why do we do denormalization in the first place?
Imagine I have my dataset transforms that do normalization and I want to see colors distribution after normalization (to see how close they are to zero mean / unit std). But instead we are forcing users to 8-bit images. There are uint16 images, there are SAR images with full fp32 range.
They only way we need denormalization, IMO, is visualization of samples. And here, if no mean/std or custom denormalization function is provided we can do scaling to 0..255 range with a warning message that results may look odd.

Reply:
Warning sounds good